Update create-buckets.mdx (#19436)
cloudflare/cloudflare-docs: src/content/docs/r2/buckets/create-buckets.mdx

:::note

> Bucket names must begin with a letter, and can only contain lowercase letters (a-z), numbers (0-9), and hyphens (-).

Reviewer: ```suggestion
- Bucket names can only contain lowercase letters (a-z), numbers (0-9), and hyphens (-).
- Bucket names cannot begin or end with a hyphen.
```
